src/store/CategoricalDisplayStore.js

```diff
@@ -320,11 +330,30 @@ export default class CategoricalDisplayStore {
         this.scrollPanel =  element;
     }
 
+    @action.bound handleTitleClick(categoryId) {
+        this.activeCategories = [];
+        for (const item of this.filteredItems) {
+            if (item.categoryId === categoryId) {
+                item.active = !item.active;
+            }
+
+            if (item.active) {
+                this.activeCategories.push(item.categoryId);
```

Comment:
the `activeCategories` use when we build the `filteredItems` items, and we pass `active` to `filteredItems` itself so that it is easy to handle the active item in template